packages/asset/contracts/Asset.sol

```diff
@@ -52,7 +51,7 @@ contract Asset is
         __ERC1155Supply_init();
         __ERC2771Handler_initialize(forwarder);
         __ERC1155Burnable_init();
-        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _grantRole(DEFAULT_ADMIN_ROLE, assetAdmin);
 
```

Comment:
Is is possible we are missing an init for ERC1155Upgradeable.sol ?

Reply:
Also does it need AccessControl init ?

Reply:
I think we don't need it, the `ERC1155URIStorageUpgradeable ` initialization sets the base URI to an empty string and we manually do that in line 49 by calling ` _setBaseURI(baseUri);`

Do you think its needed?

Reply:
I see, we are setting the baseUri to ipfs. All good